src/client/client.cpp

```diff
@@ -1897,39 +1897,36 @@ float Client::getCurRate()
 
 void Client::makeScreenshot()
 {
-	irr::video::IVideoDriver *driver = m_rendering_engine->get_video_driver();
-	irr::video::IImage* const raw_image = driver->createScreenShot();
+	video::IVideoDriver *driver = m_rendering_engine->get_video_driver();
+	video::IImage* const raw_image = driver->createScreenShot();
 
 	if (!raw_image)
 		return;
 
 	const struct tm tm = mt_localtime();
 
-	char timetstamp_c[64];
-	strftime(timetstamp_c, sizeof(timetstamp_c), "%Y%m%d_%H%M%S", &tm);
+	char timestamp_c[64];
+	strftime(timestamp_c, sizeof(timestamp_c), "%Y%m%d_%H%M%S", &tm);
 
-	std::string screenshot_dir;
-
-	if (fs::IsPathAbsolute(g_settings->get("screenshot_path")))
-		screenshot_dir = g_settings->get("screenshot_path");
-	else
-		screenshot_dir = porting::path_user + DIR_DELIM + g_settings->get("screenshot_path");
+	std::string screenshot_dir = g_settings->get("screenshot_path");
+	if (!fs::IsPathAbsolute(screenshot_dir))
+		screenshot_dir = porting::path_user + DIR_DELIM + screenshot_dir;
 
 	std::string filename_base = screenshot_dir
 			+ DIR_DELIM
 			+ std::string("screenshot_")
-			+ std::string(timetstamp_c);
+			+ timestamp_c;
 	std::string filename_ext = "." + g_settings->get("screenshot_format");
-	std::string filename;
 
 	// Create the directory if it doesn't already exist.
 	// Otherwise, saving the screenshot would fail.
-	fs::CreateDir(screenshot_dir);
+	fs::CreateAllDirs(screenshot_dir);
 
 	u32 quality = (u32)g_settings->getS32("screenshot_quality");
-	quality = MYMIN(MYMAX(quality, 0), 100) / 100.0 * 255;
+	quality = rangelim(quality, 0, 100) / 100.0f * 255;
 
-	// Try to find a unique filename
+	// Try to find an unique filename
```

Comment:
Nitpick; Actually `a unique` is correct, like `a university` because it's pronounced similar to `you` instead of a regular `u`.